core/src/main/java/io/questdb/cairo/sql/Record.java

```diff
@@ -280,6 +280,14 @@ default short getShort(int col) {
         throw new UnsupportedOperationException();
     }
 
+    default @Nullable Utf8Sequence getSplitVarcharA(int col) {
```

Comment:
We could keep things simpler without these `getSplitVarchar*` methods. Instead, certain records/functions would return `Utf8SplitString` class in `getVarchar*` calls. The class needs to be tweaked, so that it can handle the inlined case. Considering that `Utf8SplitString` is special only in `zeroPaddedSixPrefix` impl, it seems too much to create special access methods for it.

WDYT?

Reply:
I pushed one piece of the puzzle -- unifying Record.getVarchar. I'm still figuring out what to do with Memory.getVarchar.